it/test/uk/gov/hmrc/alcoholdutyaccount/controllers/PaymentsIntegrationSpec.scala

```diff
@@ -0,0 +1,341 @@
+import uk.gov.hmrc.play.bootstrap.backend.http.ErrorResponse
+
+class PaymentsIntegrationSpec extends ISpecBase with ConnectorTestHelpers {
+  protected val endpointName = "financial"
+
+  "the open payments endpoint should" should {
+    "respond with OK if able to fetch open payments" in new SetUp {
+      stubAuthorised()
+      stubGetWithParameters(url, parameters, OK, financialDataStubJson)
+
+      val response = callRoute(
+        FakeRequest("GET", routes.PaymentsController.openPayments(appaId).url)
+          .withHeaders("Authorization" -> "Bearer 12345")
+      )
+
+      status(response) shouldBe OK
+      contentAsJson(response).toString shouldBe openPayments
+
+      verifyGetWithParameters(url, parameters)
+    }
+
+    "respond with INTERNAL_SERVER_ERROR if the data retrieved cannot be parsed" in new SetUp {
+      stubAuthorised()
+      stubGetWithParameters(url, parameters, OK, "blah")
+
+      val response = callRoute(
+        FakeRequest("GET", routes.PaymentsController.openPayments(appaId).url)
+          .withHeaders("Authorization" -> "Bearer 12345")
+      )
+
+      status(response) shouldBe INTERNAL_SERVER_ERROR
+      contentAsJson(response) shouldBe Json.toJson(ErrorResponse(INTERNAL_SERVER_ERROR, "Unexpected Response"))
+
+      verifyGetWithParameters(url, parameters)
+    }
+
+    "respond with NOT_FOUND if financial data not found for appaId" in new SetUp {
+      stubAuthorised()
+      stubGetWithParameters(url, parameters, NOT_FOUND, "")
+
+      val response = callRoute(
+        FakeRequest("GET", routes.PaymentsController.openPayments(appaId).url)
+          .withHeaders("Authorization" -> "Bearer 12345")
+      )
+
+      status(response) shouldBe NOT_FOUND
+      contentAsJson(response) shouldBe Json.toJson(ErrorResponse(NOT_FOUND, "Entity not found"))
+
+      verifyGetWithParameters(url, parameters)
+    }
+
+    "respond with INTERNAL_SERVER_ERROR if error(s) returned from the financial data api call" in new SetUp {
+      stubAuthorised()
+      stubGetWithParameters(url, parameters, INTERNAL_SERVER_ERROR, "")
+
+      val response = callRoute(
+        FakeRequest("GET", routes.PaymentsController.openPayments(appaId).url)
+          .withHeaders("Authorization" -> "Bearer 12345")
+      )
+
+      status(response) shouldBe INTERNAL_SERVER_ERROR
+      contentAsJson(response) shouldBe Json.toJson(ErrorResponse(INTERNAL_SERVER_ERROR, "Unexpected Response"))
+
+      verifyGetWithParameters(url, parameters)
+    }
+  }
+
+  class SetUp {
+    val url       = config.financialDataUrl(appaId)
+
+    val parameters =     Seq(
+      "onlyOpenItems"              -> true.toString,
+      "includeLocks"               -> false.toString,
+      "calculateAccruedInterest"   -> false.toString,
+      "customerPaymentInformation" -> false.toString
+    )
+
+    val financialDataStubJson =
+      """{
+        |  "idType": "ZAD",
+        |  "idNumber": "XMADP1000800208",
```

Comment:
Could we anonymize it?

Reply:
yes, this is an oversight, it should have used generated values where possible.